closure/goog/array/array.js

```diff
@@ -346,20 +346,33 @@ goog.array.map = goog.NATIVE_ARRAY_PROTOTYPES &&
 goog.array.reduce = goog.NATIVE_ARRAY_PROTOTYPES &&
                     (goog.array.ASSUME_NATIVE_FUNCTIONS ||
                      goog.array.ARRAY_PROTOTYPE_.reduce) ?
-    function(arr, f, val, opt_obj) {
-      goog.asserts.assert(arr.length != null);
-      if (opt_obj) {
-        f = goog.bind(f, opt_obj);
+  function(arr, f, val, opt_obj) {
+    goog.asserts.assert(arr.length != null);
+    var params = [];
+    for(var i = 1, l = arguments.length; i < l; i++){
+      params.push(arguments[i]);
+    }
+    if (opt_obj) {
+      params[0] = goog.bind(f, opt_obj);
+    }
+    return goog.array.ARRAY_PROTOTYPE_.reduce.apply(arr, params);
+  } :
+  function(arr, f, val, opt_obj) {
+    goog.asserts.assert(arr.length > 0);
+    var value, length = arr.length, i = 0;
+    if (arguments.length > 2) {
+      value = arguments[2];
```

Comment:
Same issue here.  If val is undefined, it still qualifies.  If you try the changes you've suggested against any standard test of reduce, I think the specific problem will become obvious - LMK if you'd like more detail